core/src/main/java/rdx/works/core/EncryptionHelper.kt

```diff
@@ -185,3 +191,36 @@ sealed class KeySpec(val alias: String) {
     class Mnemonic(alias: String = KEY_ALIAS_MNEMONIC) : KeySpec(alias)
     class Cache(alias: String) : KeySpec(alias)
 }
+
+@Suppress("MagicNumber")
+fun String.decodeHex(): ByteArray {
+    check(length % 2 == 0) { "Must have an even length" }
+
+    return chunked(2)
+        .map { it.toInt(16).toByte() }
+        .toByteArray()
+}
+
+fun generateX25519KeyPair(): Result<Pair<String, String>> {
+    return runCatching {
+        val generator = X25519KeyPairGenerator()
+        val params = X25519KeyGenerationParameters(SecureRandom())
+        generator.init(params)
+        val keypair1: AsymmetricCipherKeyPair = generator.generateKeyPair()
+        val priv1 = keypair1.private as X25519PrivateKeyParameters
+        val pub1 = keypair1.public as X25519PublicKeyParameters
+        Pair(priv1.encoded.toHexString(), pub1.encoded.toHexString())
+    }
+}
+
+fun generateX25519SharedSecret(privateKeyCompressed: ByteArray, publicKeyCompressed: ByteArray): Result<String> {
+    return runCatching {
+        val agreement = X25519Agreement().apply {
+            init(X25519PrivateKeyParameters(privateKeyCompressed))
+        }
+
+        val secret = ByteArray(agreement.agreementSize)
+        agreement.calculateAgreement(X25519PublicKeyParameters(publicKeyCompressed), secret, 0)
+        secret.toHexString()
+    }
+}
```

Comment:
Reminder to remove this code before merging

Reply:
after using sargon for linking flow this goes away, it's just temporary for the PoC to work now.